static_src/stores/user_store.js

```diff
@@ -338,6 +355,36 @@ export class UserStore extends BaseStore {
     return role;
   }
 
+  addRoleToSavingArray(roleId) {
+    if (this._saving === null) {
+      this._saving = [];
+    }
+    this._saving.push(roleId);
+    this.emitChange();
+  }
+
+  removeRoleFromSavingArray(roleId) {
+    // These timeouts prevent a flashed text change, respective to state.
+    const minTimeForSaving = 500;
+    const minTimeForSaved = 2000;
+    let index;
+    window.setTimeout(() => {
```

Comment:
I'd look into if we could put timing related things in the JSX components rather then in the store. That way, there's no confusing about what the actual state is, the timing is at the view layer instead.